src/lua/httpc.c

```diff
@@ -155,15 +155,36 @@ luaT_httpc_request(lua_State *L)
 	if (req == NULL)
 		return luaT_error(L);
 
+	const char* content_type = NULL;
 	double timeout = TIMEOUT_INFINITY;
 	int max_header_name_length = MAX_HTTP_HEADER_NAME_LEN;
 	if (lua_isstring(L, 4)) {
 		size_t len = 0;
 		const char *body = lua_tolstring(L, 4, &len);
+		if (len > 0 && httpc_set_body(req, body, len) != 0) {
+			httpc_request_delete(req);
+			return luaT_error(L);
+                }
+        } else if ((lua_istable(L, 4)) &&
+                   (strcasecmp(method, "post") == 0)) {
```

Comment:
Shouldn't we also support the feature for `PUT` and `PATCH` HTTP request methods? (And `DELETE`? I don't remember, whether it has a body.)

Reply:
DELETE request may have a body. See [RFC 7231, Section 4.3.5](https://www.rfc-editor.org/rfc/rfc7231#section-4.3.5): 
>   A payload within a DELETE request message has no defined semantics;
>   sending a payload body on a DELETE request might cause some existing
>   implementations to reject the request.

PATCH request has a body. See [RFC 5789](https://datatracker.ietf.org/doc/html/rfc5789).

PUT request has a body, see [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#page-26).

Added PUT, PATCH and DELETE to the condition as well.

Reply:
Right, added.